lib/cosmos/tools/cmd_tlm_server/cmd_tlm_server.rb

```diff
@@ -218,16 +220,31 @@ def initialize(
           @routers.add_preidentified('PREIDENTIFIED_ROUTER', System.ports['CTS_PREIDENTIFIED'])
           @routers.add_cmd_preidentified('PREIDENTIFIED_CMD_ROUTER', System.ports['CTS_CMD_ROUTER'])
         else
+          # Create dummy interface for Replay so we can attach the preidentified routers to it.
+          # This is needed because interfaces are not mapped to targets when loading a saved_config.
+          # Since interfaces are used to access the routers, nothing is send out the preidentified
+          # interface port and TlmGrapher (most notably) does not work.
+          @replay_interface = Interface.new
+          @replay_interface.name = "REPLAY"
           @routers.all.clear unless replay_routers
-          @routers.add_preidentified('PREIDENTIFIED_ROUTER', System.ports['REPLAY_PREIDENTIFIED'])
-          @routers.add_cmd_preidentified('PREIDENTIFIED_CMD_ROUTER', System.ports['REPLAY_CMD_ROUTER'])
+          @replay_interface.routers << @routers.add_preidentified('PREIDENTIFIED_ROUTER', System.ports['REPLAY_PREIDENTIFIED'])
+          @replay_interface.cmd_routers << @routers.add_cmd_preidentified('PREIDENTIFIED_CMD_ROUTER', System.ports['REPLAY_CMD_ROUTER'])
         end
         System.telemetry.limits_change_callback = method(:limits_change_callback)
         @routers.start
 
         start(production)
       end
-    end # end def initialize
+    end
+
+    # Map any targets without interfaces to the dummy replay interface.
+    # Targets will only have an interface already mapped if the replay_routers
+    # flag was passed to the server.
+    def map_targets_to_interfaces
```

Comment:
This method should be named differently to show it is only replay mode.   Also, an attempt needs to be made to remap targets to interfaces using Interface#target_names.  Only after that fails should the target get mapped to @replay_interface.   

Overall really nice clean fix for this though.

Reply:
@ryanatball I tried to address this but never saw any existing interfaces being mapped. Not sure if this is what you wanted.